Source/UIKit/UIButton+Binding.swift

```diff
@@ -22,30 +22,37 @@
 
 import UIKit
 
+
+private var b_title_key = 0
+private var b_image_key = 0
+private var b_backgroundImage_key = 0
+
+
 public extension UIButton {
     
-    var b_title: Binding<String> {
-        return get("b_title", orSet: {
-            return Binding<String>(setter: { [weak self] value in
-                self?.setTitle(value, forState: .Normal)
-            })
-        })
+    var b_title: BindableProperty<UIButton, String?> {
+        return associatedObjectProperty(self, &b_title_key) { _ in
+            return BindableProperty(self) { control, value in
+                control.setTitle(value, for: UIControlState())
```

Comment:
For sure, its just not at all descriptive. And what is the behaviour if your enum starts in the negatives